zinc/lib/Zinc_utils.ml

```diff
@@ -19,28 +19,38 @@ module Z = struct
    fun fmt v -> Format.fprintf fmt "%s" (Z.to_string v)
 end
 
-type label = Label of string [@@deriving show {with_path = false}, eq, yojson]
+type label = int [@@deriving show {with_path = false}, eq, yojson]
+
+type variant_label = string [@@deriving show {with_path = false}, eq, yojson]
 
 module LMap = struct
-  include Map.Make (struct
-    type t = label
+  type 'a t = 'a array [@@deriving yojson]
```

Comment:
```suggestion
  type 'a t = 'a array [@@deriving show, eq, yojson]
```
if you add this you don't need to write pp, eq, or json definitions

Reply:
done